client/components/stats-date-control/stats-date-control-date-input.tsx

```diff
@@ -18,7 +21,42 @@ const DateInput: React.FC< Props > = ( {
 		onChange && onChange( event.target.value );
 	};
 
-	return <input id={ id } type="date" value={ value } onChange={ handleChange } max={ max } />;
+	const toDateString = ( date: moment.MomentInput ) => {
+		if ( moment.isMoment( date ) || moment( date ).isValid() ) {
+			return moment( date ).format( 'YYYY-MM-DD' );
+		}
+		return moment().format( 'YYYY-MM-DD' );
+	};
+
+	const handleBlur = () => {
+		if ( isNewCalendar ) {
+			const isValid = moment( value, 'YYYY-MM-DD', true ).isValid();
+			if ( ! isValid ) {
+				onChange && onChange( moment().format( 'YYYY-MM-DD' ) );
+			} else {
+				toDateString( value );
+			}
+		}
+	};
+
+	return (
+		<>
+			{ isNewCalendar ? (
+				<input
```

Comment:
Let's add a CSS class and update this selector to apply the previous styling https://github.com/Automattic/wp-calypso/blob/trunk/client/components/stats-date-control/style.scss#L121.